pyscripts/build_docker_image.py

```diff
@@ -10,193 +10,237 @@
+        parser.add_argument(
+            "--language",
+            help="language to build",
+            type=str,
+            required=True,
+            choices=all_languages,
+        )
+        parser.add_argument("--repo", help="repo with source", type=str, default=default_repo)
+        parser.add_argument(
+            "--commit", help="commit to apply (ref or branch)", type=str, default="master"
+        )
+        parser.add_argument(
+            "--variant", help="Docker image variant (blank for default)", type=str, nargs="?", const=""
+        )
+        args = parser.parse_args()
+
+        if args.repo == default_repo:
+            if args.language == "pythonpreview":
+                args.repo = "Azure/azure-iot-sdk-python-preview"
+            else:
+                args.repo = "Azure/azure-iot-sdk-" + args.language
+            print(Fore.YELLOW + "Repo not specified.  Defaulting to " + args.repo)
+
+        print_separator = "".join("/\\" for _ in range(80))
+
+        auth_config = {
+            "username": os.environ["IOTHUB_E2E_REPO_USER"],
+            "password": os.environ["IOTHUB_E2E_REPO_PASSWORD"],
+        }
+
+
+        def get_dockerfile_directory(tags):
+            script_dir = os.path.dirname(os.path.realpath(__file__))
+            return os.path.normpath(os.path.join(script_dir, "../ci-wrappers/" + tags.language))
+
+
+        def print_filtered_docker_line(lineobj):
+            lineobj = lineobj.splitlines()
+            for line in lineobj:
+                try:
+                    linecode = line.strip().decode("utf-8")
+                    linecode = linecode[:linecode.rfind("\r\n")] + '}'
+                    jline = json.loads(linecode)
+                    obj = jline["stream"]
+                except:
+                    print(''.join([chr(i) if i < 128 else '\x23' for i in line]))
                 else:
-                    pass
+                    if "status" in obj:
+                        if "id" in obj:
+                            if obj["status"] not in [
+                                "Waiting",
+                                "Downloading",
+                                "Verifying Checksum",
+                                "Extracting",
+                                "Preparing",
+                                "Pushing",
+                            ]:
+                                print("{}: {}".format(obj["status"], obj["id"]))
+                            else:
+                                pass
+                        else:
+                            print(obj["status"])
+                    elif "error" in obj:
+                        print ("docker error: {}".format(line))
+                        raise Exception(obj["error"])
+                    elif "Step" in obj or "---" in obj:
+                        print("{}".format(obj).strip())
+                    elif obj == "\n":
+                        pass
```

Comment:
Wait, is "obj" a string or an object?  With a name like "obj", I assume it's an object.  Above, you're treating it like an object, but here it looks like you're treating it like a string, but print("{}".format(obj).strip()).  makes it look like it's not a string (because you're using the roundabout "{}".format() to convert it to a string, but then you strip it, so there must be leading or trailing spaces, so it must be a string.  Help, what is it?

Reply:
fixed. handle json exception and print ascii7